### ISSUE TITLE
How to Setup Microsoft Entra ID SSO

### ISSUE THREAD

**Issue by moellert-biw** [docusealco/docuseal#351]
Hello everyone, 
I am in the process of setting up SSO using the IdP Microsoft Entra ID. Unfortunately, I have not been able to complete a successful login. I have created an app with SAML2 SSO, copied the ‘SSO Service URL’ and ‘Certificate’ from Entra ID. I also tried to change the attribute mapping for the user name from the UPN to the email address.
In the Entra ID SIgn-In logs, I can see successful login attempts as soon as I try to sign in, but DocuSeal only returns the message ‘User not found.’ when I try to log in. 
Has anyone managed to successfully set up Microsoft Entra ID as an SSO provider? What could be the possible causes of the ‘User not found.’ message?

**Comment by omohokcoj** [comment]
@moellert-biw looks like your SAML configs work - but the email of the user from Entra ID doesn't match any of the user emails in your on-premises app. Do you have a user in your on-prem app that matches the email of your Entra ID user?

**Comment by moellert-biw** [comment]
@omohokcoj I have my own user in on-prem DocuSeal with the same email in Entra ID. For testing purposes I have also tried to provision a new user (not yet created in on-prem DocuSeal). Both scenarios give the same "User not found." message. 

These are the set Claims in Entra ID:
![CleanShot 2024-09-01 at 22 37 01](https://github.com/user-attachments/assets/bd597000-fea4-4442-8532-babe7a5c180b)

**Comment by omohokcoj** [comment]
@moellert-biw try to change user email  mapping to user.userprincipalname (user.email is for the azure internal email but not the azure registered user company email)
Let me know if it solves the issue
![image](https://github.com/user-attachments/assets/7838b71e-fe42-4ec4-96a8-b208393e1395)

**Comment by moellert-biw** [comment]
@omohokcoj I have changed the claims like this:
![CleanShot 2024-09-02 at 08 33 41@2x](https://github.com/user-attachments/assets/230d4c89-6297-41dc-ab0c-e47bf061b67e)

I hope these are the right changes.

Unfortunately the "User not found." message still pops up.

Side note: In our tenant the UPN is not the primary SMTP address. We have a policy that sets the primary SMTP address to surname@domain.com whereas the UPN is always like givenname.surname@domain.com 
I have tested to set the username in on-prem DocuSeal to both givenname.surname@domain.com and surname@domain.com
Unfortunately both tests lead to the "User not found." message.

**Comment by omohokcoj** [comment]
@moellert-biw the last time we set up SAML with Entra ID - it worked when using those instructions:
https://stackoverflowteams.help/en/articles/4538487-configure-single-sign-on-sso-with-entra-id 
Could you please check it and see if maybe some configurations in your Entra ID are missing?

**Comment by moellert-biw** [comment]
@omohokcoj First of all, thank you for your help!
I have read the linked article and checked the settings. I am unsure about the `Basic SAML Configuration`. This is the current configuration:
![CleanShot 2024-09-02 at 13 32 53](https://github.com/user-attachments/assets/99c19c12-d944-4a07-b02f-12dd7eb35bb8)

I have copied the `Login URL` from Entra to DocuSeal field `SSO Service URL`

As suggested in https://www.docuseal.co/docs/configuring-saml-with-google-workspace I have set the `Entity ID` to the DocuSeal `Metadata URL`. The stackoverflow article does not suggest to use the `Metadata URL` but a string of text.

**Comment by omohokcoj** [comment]
@moellert-biw did it fix the issue?

**Comment by moellert-biw** [comment]
@omohokcoj unfortunately no, still seeing the same "User not found." message

**Comment by omohokcoj** [comment]
@moellert-biw could you please try to change NameID matching to nameid-format:emailAddress

![image](https://github.com/user-attachments/assets/765abf24-5481-429c-9c1c-03affc6096a6)
![image](https://github.com/user-attachments/assets/ba215c5d-b1d4-45a1-9a42-ee4842666e4e)

**Comment by moellert-biw** [comment]
@omohokcoj I have changed the NameID according to your screenhsots:
![CleanShot 2024-09-02 at 22 34 41](https://github.com/user-attachments/assets/8ca4b4c9-4ed1-4dce-b3b7-b2daed4e4df0)
![CleanShot 2024-09-02 at 22 37 33](https://github.com/user-attachments/assets/cdb36e8d-30ba-48c7-b40d-79474be175f5)

Unfortunately still seeing "User not found." on login

**Comment by omohokcoj** [comment]
@moellert-biw could you please update to the latest Pro version and try again - the error message should contain your `nameid` now which might help to debug the issue

**Comment by moellert-biw** [comment]
@omohokcoj I have upgraded to the latest version and indeed the new "User not found" message contains `nameid` 
It is `Surname@domain.com` (first letter is uppercase) instead of `surname@domain.com`
Maybe there is an issue with case sensitivity?

As a workaround I have tried to change the username in on-prem DocuSeal to `Surname@domain.com` but as soon as I hit the Update button (Settings > Profile) it is all lowercase again.
Also tried to change the first letter using Settings > Users but this is saving the first letter lowercase as well.
